### PR TITLE
wxWidgets 3.1.3

### PR DESCRIPTION
**Pull request by alarcher**
Removed files in /usr/share and symlink in bin that conflict with version 2.8.
Audacity needs 2.8 so we need to provide both for some time.

### CODE REVIEW

**Comment by pyhalov** [comment]
Have you already used it somewhere?
Do we really need 32-bit version?

**Comment by alarcher** [comment]
I have used 3.0.3 so this is not to merge right away, need some testing. I'd like your opinion on managing both versions: do we provide only /usr/share files from 2.8 for now?

**Comment by pyhalov** [comment]
We have aclocal file and some bakefiles. Aclocal file is not critical (however, it's pleasant to have), but I don't know where are bakefiles used.

**Comment by pyhalov** [comment]
@alarcher Can we update and merge this PR?

**Comment by alarcher** [comment]
I can rebase and merge.

**Comment by pyhalov** [comment]
Can you bump it?

**Comment by alarcher** [comment]
If it is easy yes otherwise no, I cannot spend much time on it.

**Comment by Mno-hime** [comment]
I am having a look at v3.1.2.

**Comment by Mno-hime** [comment]
I guess I need 64-bit `library/desktop/webkitgtk2` first:
```
/usr/gcc/6/bin/g++ -shared -fPIC -o webkit2_extu-3.1.2.so webkit2_ext_webview_webkit2_extension.o  -pthreads -m64    -lwebkit2gtk-4.0 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -ljavascriptcoregtk-4.0 -lglib-2.0 -lz -lsocket -lsecret-1 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -llzma -lm
ld: fatal: library -lwebkit2gtk-4.0: not found
ld: fatal: library -ljavascriptcoregtk-4.0: not found
ld: fatal: file processing errors. No output written to webkit2_extu-3.1.2.so
collect2: error: ld returned 1 exit status
```

**Comment by Mno-hime** [comment]
How can it not resolve `libGL.so.1`?
```
[wxwidgets3] {build-userland} newman:~/ws/oi-userland/components/library/wxwidgets-3 $ gmake MAKEFILE_PREREQ= REQUIRED_PACKAGES
/usr/gnu/bin/make RESOLVE_DEPS= /export/home/newman/ws/oi-userland/components/library/wxwidgets-3/build/.resolved-i386
make[1]: Entering directory '/export/home/newman/ws/oi-userland/components/library/wxwidgets-3'
/usr/bin/pkgdepend resolve  -m /export/home/newman/ws/oi-userland/components/library/wxwidgets-3/build/manifest-i386-wxwidgets-3.depend
/export/home/newman/ws/oi-userland/components/library/wxwidgets-3/build/manifest-i386-wxwidgets-3.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=libGL.so.1 \
        pkg.debug.depend.reason=usr/lib/libwx_gtk3u_gl-3.1.so.2 \
        pkg.debug.depend.type=elf \
        pkg.debug.depend.path=lib \
        pkg.debug.depend.path=usr/gcc/6/lib \
        pkg.debug.depend.path=usr/lib'.
/export/home/newman/ws/oi-userland/components/library/wxwidgets-3/build/manifest-i386-wxwidgets-3.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=libGL.so.1 \
        pkg.debug.depend.reason=usr/lib/amd64/libwx_gtk3u_gl-3.1.so.2 \
        pkg.debug.depend.type=elf \
        pkg.debug.depend.path=lib/64 \
        pkg.debug.depend.path=usr/gcc/6/lib/amd64 \
        pkg.debug.depend.path=usr/lib/64'.
make[1]: *** [/export/home/newman/ws/oi-userland/make-rules/ips.mk:405: /export/home/newman/ws/oi-userland/components/library/wxwidgets-3/build/.resolved-i386] Error 1
make[1]: Leaving directory '/export/home/newman/ws/oi-userland/components/library/wxwidgets-3'
gmake: *** [/export/home/newman/ws/oi-userland/make-rules/ips.mk:418: REQUIRED_PACKAGES] Error 2

lrwxrwxrwx   1 root     root          38 May 15  2018 /usr/lib/GL/libGL.so.1 -> ../../../var/run/opengl/lib/libGL.so.1*
lrwxrwxrwx   1 root     root          13 May 15  2018 /usr/lib/libGL.so.1 -> GL/libGL.so.1*
lrwxrwxrwx   1 root     root          14 May 15  2018 /usr/lib/mesa/libGL.so.1 -> libGL.so.1.2.0*
-r-xr-xr-x   1 root     bin         604K Jan 11  2019 /usr/lib/mesa/libGL.so.1.2.0*
lrwxrwxrwx   1 root     root          35 Jul 21 13:57 /var/run/opengl/lib/libGL.so.1 -> ../../../../usr/lib/mesa/libGL.so.1*

```

**Comment by Mno-hime** [comment]
[64-bit webkitgtk2](https://github.com/OpenIndiana/oi-userland/pull/5143) will is needed.

Although 3.1.2 is "development" version, upstream suggests it to be used https://www.wxwidgets.org/news/2018/12/wxwidgets-3.1.2-released/ & https://github.com/wxWidgets/wxWidgets/releases/tag/v3.1.2.

**Comment by pyhalov** [comment]
This is usual issue. GL.so.1 is a link created at boot,  pkgdepend can't handle it (it doesn't depend to any package). Usual practice - ignore dependency and depend on ogl-select.

**Comment by Mno-hime** [comment]
Updated to v3.1.2. The component is GTK3-only as it needs to be build twice for dual GTK support (as in https://src.fedoraproject.org/rpms/wxGTK3/blob/master/f/wxGTK3.spec#_276).

**Comment by pyhalov** [comment]
```
reating Plan (Checking for conflicting actions): -
pkg install: The following packages deliver conflicting action types to usr/bin/wxrc:

  file:
    pkg://openindiana.org/library/graphics/wxwidgets@2.8.12,5.11-2018.0.0.1:20180524T210016Z
  link:
    pkg://userland/library/graphics/wxwidgets-3@3.1.2,5.11-2018.0.0.0:20190809T071755Z

These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

The requested change to the system attempts to install multiple actions
for link 'usr/bin/wx-config' with conflicting attributes:

    1 package delivers 'link path=usr/bin/wx-config target=../lib/wx/config/gtk2-unicode-release-2.8':
        pkg://openindiana.org/library/graphics/wxwidgets@2.8.12,5.11-2018.0.0.1:20180524T210016Z
    1 package delivers 'link path=usr/bin/wx-config target=../lib/wx/config/gtk3-unicode-3.1':
        pkg://userland/library/graphics/wxwidgets-3@3.1.2,5.11-2018.0.0.0:20190809T071755Z


These packages cannot be installed together. Any non-conflicting subset
of the above packages can be installed.

The requested change to the system attempts to install multiple actions
for link 'usr/bin/amd64/wx-config' with conflicting attributes:

    1 package delivers 'link path=usr/bin/amd64/wx-config target=../../lib/amd64/wx/config/gtk2-unicode-release-2.8':
        pkg://openindiana.org/library/graphics/wxwidgets@2.8.12,5.11-2018.0.0.1:20180524T210016Z
    1 package delivers 'link path=usr/bin/amd64/wx-config target=../../lib/amd64/wx/config/gtk3-unicode-3.1':
        pkg://userland/library/graphics/wxwidgets-3@3.1.2,5.11-2018.0.0.0:20190809T071755Z

```
Should likely rename it to wx-config-3

File: components/library/wxwidgets-3/wxwidgets-3.p5m [hunk 1]
```diff
@@ -26,11 +26,11 @@ license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
...
 <transform file -> add pkg.depend.bypass-generate libGL\.so\.1>
 
-link path=usr/bin/$(MACH64)/wx-config \
+link path=usr/bin/$(MACH64)/wx-config-3 \
     target=../../lib/$(MACH64)/wx/config/gtk3-unicode-3.1
 link path=usr/bin/$(MACH64)/wxrc target=wxrc-3.1
 file path=usr/bin/$(MACH64)/wxrc-3.1
-link path=usr/bin/wx-config target=../lib/wx/config/gtk3-unicode-3.1
+link path=usr/bin/wx-config-3 target=../lib/wx/config/gtk3-unicode-3.1
 link path=usr/bin/wxrc target=wxrc-3.1
```
Comment (pyhalov) [inline components/library/wxwidgets-3/wxwidgets-3.p5m]: The same is true for wxrc. 
Try to install wxWidgets-3 and old wxwidgets together.

Comment (Mno-hime) [reply components/library/wxwidgets-3/wxwidgets-3.p5m]: With the latest fix I was able to install `wxwidgets` and `wxwidgets-3` together. `pkg fix` on both packages is clean.

Comment (pyhalov) [reply components/library/wxwidgets-3/wxwidgets-3.p5m]: OK. Have you compiled anything against it (and used compiled app)? 
I'm trying to build updated audacity, but it has own issues (unrelated to wxwidgets).

Comment (Mno-hime) [reply components/library/wxwidgets-3/wxwidgets-3.p5m]: Right. I think someone on the oi-discuss ml was building (older version of) Filezilla with wxwidgets, might be a good candidate. Will have a look during the weekend.

**Comment by pyhalov** [comment]
Tested it with updated Audacity. Seems to be working (given that audacity itself crashes now and then, hard to tell for sure) :)

**Comment by Mno-hime** [comment]
FileZilla does not support "unstable" 3.1.x version, requires 3.0.x. If patched out it fails like this:

```
In file included from /usr/include/wx-3.1/wx/filesys.h:26:0,
                 from /usr/include/wx-3.1/wx/xrc/xmlres.h:23,
                 from /export/home/newman/ws/oi-userland/components/desktop/filezilla/filezilla-3.44.1/src/interface/filezilla.h:75:
/usr/include/wx-3.1/wx/filename.h: In static member function 'static wxUniChar wxFileName::GetPathSeparator(wxPathFormat)':
/usr/include/wx-3.1/wx/filename.h:460:43: error: ambiguous overload for 'operator[]' (operand types are 'wxString' and 'unsigned int')
         { return GetPathSeparators(format)[0u]; }
                                           ^
/usr/include/wx-3.1/wx/filename.h:460:43: note: candidate: operator[](const wchar_t*, long int) <built-in>
/usr/include/wx-3.1/wx/filename.h:460:43: note: candidate: operator[](const char*, long int) <built-in>
```

Haven't look if this particular one is easily fixable.

libfilezilla and FileZilla components: https://github.com/Mno-hime/oi-userland/tree/filezilla.

**Comment by alarcher** [comment]
@Mno-hime Merge wxwidget, we can fix the compatbility in filezilla.

**Comment by Mno-hime** [comment]
I'll bump to 3.1.3 first.

**Pull request merged**